lib/bridge/instance.js

```diff
@@ -0,0 +1,175 @@
+/*eslint no-invalid-this: 0*/ // eslint doesn't understand Promise.coroutine wrapping
+"use strict";
+var Promise = require("bluebird");
+var extend = require("extend");
+
+var store = require("../store");
+var ircToMatrix = require("./irc-to-matrix.js");
+var matrixToIrc = require("./matrix-to-irc.js");
+var membershiplists = require("./membershiplists.js");
+var ircLib = require("../irclib/irc.js");
+var names = require("../irclib/names.js");
+var IrcServer = require("../irclib/server.js").IrcServer;
+var matrixLib = require("../mxlib/matrix");
+var MatrixUser = require("../models/users").MatrixUser;
+var logging = require("../logging");
+var log = logging.get("IrcBridge");
+var Bridge = require("matrix-appservice-bridge").Bridge;
+
+const DELAY_TIME_MS = 10 * 1000;
+const DEAD_TIME_MS = 60 * 1000 * 5; // 5min
+
+
+function IrcBridge(config, registration) {
+    this.config = config;
+    this.registration = registration;
+    this.ircServers = [];
+    this.bridge = null; // Bridge
+}
+
+IrcBridge.prototype.run = Promise.coroutine(function*(port) {
+    // connect to the DB, blow away any old config mappings, we're setting new ones now.
+    yield store.connectToDatabase(this.config.ircService.databaseUri);
+    yield store.rooms.removeConfigMappings();
+
+    // maintain a list of IRC servers in-use
+    let serverDomains = Object.keys(this.config.ircService.servers);
+    for (var i = 0; i < serverDomains.length; i++) {
```

Comment:
Can't really do this *nicely*. `forEach` means a new `function` is required, and inside the `for` we `yield`, so you need to `forEach` a generator function. You can't even do arrow functions `*() =>` with generators :/ and need to IIFE the function `Promise.coroutine()` returns. In all, it looks terrible and is confusing.